src/Logging.W3C/src/W3CLogger.cs

```diff
@@ -0,0 +1,166 @@
+using System;
+using Microsoft.Extensions.Options;
+using System.Collections.Generic;
+using Microsoft.AspNetCore.Http;
+using System.Numerics;
+using System.Text;
+using Microsoft.Net.Http.Headers;
+using System.Text.RegularExpressions;
+using System.Globalization;
+
+namespace Microsoft.Extensions.Logging.W3C
+{
+    internal sealed class W3CLogger : ILogger, IDisposable
+    {
+        private readonly string _name;
+        private readonly W3CLoggerProcessor? _messageQueue;
+        private readonly IOptionsMonitor<W3CLoggerOptions> _options;
+        private readonly bool _isActive;
+        private readonly W3CLoggingFields _loggingFields;
+
+        internal W3CLogger(string name, IOptionsMonitor<W3CLoggerOptions> options)
+        {
+            if (name == null)
+            {
+                throw new ArgumentNullException(nameof(name));
+            }
+
+            _name = name;
+            _options = options;
+
+            // If the info isn't coming from the _w3cLogger in HttpLoggingMiddleware, don't log anything
+            if (name == "Microsoft.AspNetCore.W3CLogging")
```

Comment:
TODO - I use this arbitrary string as the category name for W3CLogging in various places - would it be better to use some constant? nameof({some class that HttpLoggingMiddleware & Logging.W3C know about})?